localstack/services/dynamodb/provider.py

```diff
@@ -433,9 +433,15 @@ def delete_expired_items() -> int:
                     items_to_delete = result.get("Items", [])
                     no_expired_items += len(items_to_delete)
                     table_description = client.describe_table(TableName=table_name)
-                    partition_key = _get_partition_key(table_description)
+                    partition_key, range_key = _get_hash_and_range_key(table_description)
                     keys_to_delete = [
-                        {partition_key: item.get(partition_key)} for item in items_to_delete
+                        {partition_key: item.get(partition_key)}
+                        if range_key is None
```

Comment:
This might need additional condition for `partition_key` if following suggestion is accepted.